activity-handler.js

```diff
@@ -0,0 +1,48 @@
+"use strict";
+
+var os = require("os"),
+    util = require("util");
+
+var async = require("async"),
+    debug = require("debug")("swfr:activity"),
+    humanize = require("humanize-plus");
+
+var activity = require("./activity"),
+    activities = require("./lib/activities");
+
+var activityHandler = function activityHandler(activitiesFolder, id) {
+  id = id | 1;
+
+  var taskContext = this;
+  var lookup = activities(activitiesFolder);
+
+  return function(task, callback) {
+    console.log("worker #%d:", id, task);
+    var name = task.activityType.name,
+        version = task.activityType.version,
+        activity = lookup(name, version);
+
+    if (!activity) {
+      return callback(new Error(util.format("Unable to locate activity: %s@%s", name, version)));
+    }
+
+    return activity.apply(taskContext, task.input.concat(function(err, result) {
```

Comment:
...which makes no sense because the activity should have no notion of the decider's context.  `SyncDeciderContext` should probably call activity handlers w/o any context.